### ISSUE TITLE
ajax error

### ISSUE THREAD

**Issue by Lepelaar** [TypeRocket/core#77]
Out of nowhere I got this problem with a part of my script that has been working fine for weeks.
An ajax error is occurring: "Your request had an error. 404 - Not Found" and I have no idea how this can happen or what is causing this.

It happen in an admin form that submits some value's to the options table. This was tested and has been working fine, but today I installed a new environment (vagrant) and cloned the lastest TypeRocket (as MU-plugin). I put my own scripts in and tried to use the form to setup some variables in the options table. But I got this error as mentioned and the inspecting the browser I find that this is caused bu URL: https://one.wordpress.test/tr_json_api/v1/option/

I made sure that all the files involved are in the right place and have no errors.  I hHave no clue what is happening, so hopefully you could give me some pointers how to approach this problem. 

Thanks in advance.

**Comment by kevindees** [comment]
Hey @Lepelaar 

have you flushed your permalinks and turned on friendly URLs?

**Comment by Lepelaar** [comment]
Hi Kevin,

Thanks for replying. Yes, I did check that right away, it was the first (and only) idea I had :-\

I tried investigating the code and I think the 404 comes from the URL not being completed.
In `Form.php` (\typerocket\vendor\typerocket\core\src\Elements\) in the `useJson` method, it looks like `$this->itemId` should be appended, but it is empty. Can't figure out why, nor what it should be.
But what baffles me most is that everything was working fine before.

Any other ideas?

Oh and BTW I really like TypeRocket a lot! Thank you for sharing it with us!

**Comment by kevindees** [comment]
Hey @Lepelaar 

I just made an update to `useJson()` maybe the new features will help you?

https://github.com/TypeRocket/core/releases/tag/v4.0.77

Thanks,
Kevin

**Comment by Lepelaar** [comment]
Thanx a bunch Kevin! 
I will try it out first thing on Monday when I am back at work.
Have a great weekend!

**Comment by Lepelaar** [comment]
Hi Kevin,
Tried out the updated file this morning; unfortunately that made no difference.
After that I replaced the whole mu-plugins folder from an old backup (with the very first version that had the error causing script). And that too gave now an error. So it was pretty clear that the problem was not TypeRocket.
I ended up doing a complete new provisioning of the VM and that solved the problem (or rather not solved but fixed, because it is still completely unclear to me what caused it)
Anyways, I am up and running again :-)

Thank you so much for the support!
Greetz,
Lepelaar
